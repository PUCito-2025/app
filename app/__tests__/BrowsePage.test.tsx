import { render } from "@testing-library/react";
import { expect, test } from "vitest";

import BrowsePage from "@/app/browse/page";

test("BrowsePage matches snapshot", () => {
  const { container } = render(<BrowsePage />);
  expect(container).toMatchSnapshot();
});
