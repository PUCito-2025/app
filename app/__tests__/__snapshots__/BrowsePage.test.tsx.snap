// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BrowsePage matches snapshot 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col"
  >
    <div
      class="flex-grow px-4 py-6 lg:px-10"
    >
      <div
        class="flex flex-wrap items-center gap-3 text-sm"
      >
        <div
          class="rounded-full bg-blue-100 px-3 py-1 text-blue-700"
        >
          $5M - $15M CLP
        </div>
        <div
          class="rounded-full bg-blue-100 px-3 py-1 text-blue-700"
        >
          Santiago
        </div>
        <div
          class="rounded-full bg-blue-100 px-3 py-1 text-blue-700"
        >
          Toyota
        </div>
        <button
          class="ml-2 text-sm text-red-500 underline"
          type="button"
        >
          Clear All
        </button>
      </div>
      <div
        class="mt-6 flex gap-6"
      >
        <aside
          class="hidden w-64 lg:block"
        >
          <div
            class="space-y-6 text-sm"
          >
            <div>
              <h4
                class="mb-2 font-semibold"
              >
                Price Range
              </h4>
              <div
                class="flex gap-2"
              >
                <input
                  class="w-full rounded border p-1"
                  placeholder="Min"
                />
                <input
                  class="w-full rounded border p-1"
                  placeholder="Max"
                />
              </div>
            </div>
            <div>
              <h4
                class="mb-2 font-semibold"
              >
                Location
              </h4>
              <select
                class="w-full rounded border p-2"
              >
                <option>
                  All Regions
                </option>
              </select>
            </div>
            <div>
              <h4
                class="mb-2 font-semibold"
              >
                Brand & Model
              </h4>
              <input
                class="w-full rounded border p-2"
                placeholder="Search brands..."
              />
              <select
                class="mt-2 w-full rounded border p-2"
              >
                <option>
                  All Models
                </option>
              </select>
            </div>
            <div>
              <h4
                class="mb-2 font-semibold"
              >
                Year
              </h4>
              <select
                class="w-full rounded border p-2"
              >
                <option>
                  Any Year
                </option>
              </select>
            </div>
            <div>
              <h4
                class="mb-2 font-semibold"
              >
                Transmission
              </h4>
              <select
                class="w-full rounded border p-2"
              >
                <option>
                  Any Transmission
                </option>
              </select>
            </div>
            <div>
              <h4
                class="mb-2 font-semibold"
              >
                Fuel Type
              </h4>
              <select
                class="w-full rounded border p-2"
              >
                <option>
                  Any Fuel Type
                </option>
              </select>
            </div>
          </div>
        </aside>
        <main
          class="flex-1"
        >
          <div
            class="mb-4 flex items-center justify-between"
          >
            <span
              class="text-sm text-gray-600"
            >
              Sort by:
            </span>
            <select
              class="rounded-md border p-2 text-sm"
            >
              <option>
                Relevance
              </option>
              <option>
                Price: Low to High
              </option>
              <option>
                Price: High to Low
              </option>
              <option>
                Newest
              </option>
            </select>
          </div>
          <div
            class="grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
          >
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
            <div
              class="relative rounded-md border bg-white p-3"
            >
              <div
                class="mb-3 h-36 rounded bg-gray-200"
              />
              <h3
                class="text-sm font-semibold"
              >
                Toyota Corolla 2020
              </h3>
              <p
                class="text-sm font-bold text-blue-600"
              >
                CLP $12,500,000
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Santiago, Chile • 45,000 km
              </p>
              <p
                class="text-xs text-gray-500"
              >
                Manual, Petrol
              </p>
              <div
                class="mt-3 flex items-center justify-between"
              >
                <button
                  class="text-sm font-medium text-blue-600 hover:underline"
                  type="button"
                >
                  Compare
                </button>
                <button
                  class="rounded-md bg-blue-600 px-3 py-1 text-sm text-white hover:bg-blue-700"
                  type="button"
                >
                  View Details
                </button>
              </div>
              <div
                class="absolute top-2 right-2 h-5 w-5 rounded-full bg-red-300"
              />
            </div>
          </div>
          <div
            class="mt-8 flex justify-center"
          >
            <div
              class="flex items-center gap-2 text-sm"
            >
              <button
                class="rounded-md border px-3 py-1"
                type="button"
              >
                1
              </button>
              <button
                class="rounded-md border px-3 py-1"
                type="button"
              >
                2
              </button>
              <button
                class="rounded-md border px-3 py-1"
                type="button"
              >
                3
              </button>
            </div>
          </div>
        </main>
      </div>
    </div>
    <footer
      class="bg-blue-950 px-6 py-6 text-white"
    >
      <div
        class="mx-auto grid max-w-7xl grid-cols-1 gap-8 text-xs md:grid-cols-4"
      >
        <div>
          <h3
            class="mb-2 font-bold"
          >
            PUCito
          </h3>
          <p>
            Your trusted platform for studying across Canvas.
          </p>
          <div
            class="mt-3 flex gap-3"
          >
            <div
              class="h-5 w-5 rounded-full bg-white"
            />
            <div
              class="h-5 w-5 rounded-full bg-white"
            />
            <div
              class="h-5 w-5 rounded-full bg-white"
            />
          </div>
        </div>
        <div>
          <h4
            class="mb-2 font-semibold"
          >
            Support
          </h4>
          <ul
            class="space-y-1"
          >
            <li>
              Help Center
            </li>
            <li>
              Contact Us
            </li>
            <li>
              Privacy Policy
            </li>
            <li>
              Terms of Service
            </li>
          </ul>
        </div>
        <div>
          <h4
            class="mb-2 font-semibold"
          >
            Connect
          </h4>
          <ul
            class="space-y-1"
          >
            <li>
              Instagram
            </li>
            <li>
              Facebook
            </li>
            <li>
              Twitter
            </li>
          </ul>
        </div>
      </div>
      <div
        class="mt-4 text-center text-xs opacity-70"
      >
        © 2025 PUCito. All rights reserved.
      </div>
    </footer>
  </div>
</div>
`;
